src/rt-api/post-contact.ts

```diff
@@ -90,42 +119,78 @@ export const postContact = async (
     });
   }
 
-  if (contact.contactFieldsToDelete){
-    console.log(contact.contactFieldsToDelete)
-    if(contact.contactFieldsToDelete.emailAddresses){
-      for(const internetRecId of contact.contactFieldsToDelete.emailAddresses){
-        const internetResult = await Axios({
-          method: "post",
+  // If present, update contact's url addresses
+  if (contact.Urls) {
+    contact.Urls.forEach(async (url: UrlUpdate) => {
+      try {
+        const method = url.id === 0 ? "post" : "put";
+        const urlUrl =
+          url.id === 0
+            ? `/contacts/${contact.ContactRecord.id}/urls`
+            : `/contacts/${contact.ContactRecord.id}/urls/${url.id}`;
+        if (url.id === 0) {
+          delete url.id;
+        }
+
+        const urlResult = await Axios({
+          method,
+          url: REDTAIL_TWAPI_URL + urlUrl,
+          headers: createRtApiHeaders(userKey),
+          data: url,
+        });
+        if (urlResult.status !== 200) {
+          return 5;
+        }
+      } catch (e) {
+        logger.error("Url RT API PUT error: " + JSON.stringify(e));
+        return 5;
+      }
+    });
+  }
```

Comment:
After seeing the same basic function of updating/creating contact fields 4 times in a row I wonder if we can extract this into a single function in a new file.. Like `CreateOrUpdateContactFields`, params would be type of field for URL and for checks for country code etc and then the data itself. Thoughts?